aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequestFactory.java

```diff
@@ -221,6 +222,14 @@ private AppSyncGraphQLRequestFactory() { }
     public static <T extends Model> GraphQLRequest<T> buildSubscription(
             Class<T> modelClass,
             SubscriptionType type
+    ) throws ApiException {
+        return buildSubscription(modelClass, type, null);
+    }
+
+    public static <T extends Model> GraphQLRequest<T> buildSubscription(
+            Class<T> modelClass,
+            SubscriptionType type,
+            CognitoUserPoolsAuthProvider cognitoAuth
```

Comment:
Would there be other overloaded versions of `buildSubscription` with other AuthTypes such as IAM, APIKey and OIDC?

Reply:
We're going to do a deep dive with Richard on this after Christmas break but for now are just handling the simple case of doing this with Cognito since that's the only provider the documentation talks about: 

```
Passing in the current user can be done dynamically in your code by using Auth.currentAuthenticatedUser() in JavaScript, AWSMobileClient.default().username in iOS, or AWSMobileClient.getInstance().getUsername() in Android.
```